packages/frontend/component/src/components/overlay-modal/overlay-modal.tsx

```diff
@@ -0,0 +1,82 @@
+import { DialogTrigger } from '@radix-ui/react-dialog';
+import { memo, useCallback, useEffect } from 'react';
+
+import { Button, type ButtonProps } from '../../ui/button';
+import { Modal, type ModalProps } from '../../ui/modal';
+import * as styles from './overlay-modal.css';
+
+const defaultContentOptions: ModalProps['contentOptions'] = {
+  style: {
+    padding: 0,
+    overflow: 'hidden',
+    boxShadow: 'var(--affine-menu-shadow)',
+  },
+};
+const defaultOverlayOptions: ModalProps['overlayOptions'] = {
+  style: {
+    background: 'var(--affine-white-80)',
+    backdropFilter: 'blur(2px)',
+  },
+};
+
+export interface OverlayModalProps extends ModalProps {
```

Comment:
I think it's better to provide a `to` prop, to make the confirm button behave as a `<Link />`. Also we don't need to handle the open url behavior outside component.

Reply:
I will add `to` prop, but I will keep the `onConfirm` in order to pass in different click events.

Reply:
For example, click to open enable cloud modal